Segment-ComScore/Classes/SEGComScoreIntegration.m

```diff
@@ -441,34 +391,28 @@ - (void)videoContentPlaying:(NSDictionary *)properties withOptions:(NSDictionary
     // we need to call setAsset with the content metadata.  If ns_st_ad is not present, that means the last
     // observed event was related to content, in which case a setAsset call should not be made (because asset
     // did not change).
-    SCORStreamingPlaybackSession *session = [self.streamAnalytics playbackSession];
-    SCORStreamingAsset *asset = [session asset];
-    if ([asset containsLabel:@"ns_st_ad"]) {
-        [[self.streamAnalytics playbackSession] setAssetWithLabels:map];
-        SEGLog(@"[[SCORStreamingAnalytics playbackSession] setAssetWithLabels:%@]", map);
-    }
-
-    if ([properties[@"position"] longValue]) {
-        long playPosition = [properties[@"position"] longValue];
-        [self.streamAnalytics notifyPlayWithPosition:playPosition];
-        SEGLog(@"[[SCORStreamingAnalytics streamAnalytics] notifyPlayWithPosition:%ld]", playPosition);
-    } else {
-        [self.streamAnalytics notifyPlay];
-        SEGLog(@"[[SCORStreamingAnalytics streamAnalytics] notifyPlay]");
+    
+    SCORStreamingContentMetadata *contentMetaData = [self instantiateContentMetaData:map];
+    
+    SCORStreamingAdvertisementMetadata * advertisingMetaData = [SCORStreamingAdvertisementMetadata advertisementMetadataWithBuilderBlock:^(SCORStreamingAdvertisementMetadataBuilder *builder) {
+        [builder setMediaType: SCORStreamingAdvertisementTypeOnDemandPreRoll];
+        [builder setRelatedContentMetadata: contentMetaData];
+    }];
+    
+    if ([map objectForKey:@"ns_st_ad"]) {
+        [self.streamAnalytics setMetadata:advertisingMetaData];
     }
+   
+    [self movePosition:properties];
+    [self.streamAnalytics notifyPlay];
 }
 
 
 - (void)videoContentCompleted:(NSDictionary *)properties withOptions:(NSDictionary *)integrations
 {
-    if ([properties[@"position"] longValue]) {
-        long playPosition = [properties[@"position"] longValue];
-        [self.streamAnalytics notifyEndWithPosition:playPosition];
-        SEGLog(@"[[SCORStreamingAnalytics streamAnalytics] notifyEndWithPosition:%ld]", playPosition);
-    } else {
-        [self.streamAnalytics notifyEnd];
-        SEGLog(@"[[SCORStreamingAnalytics streamAnalytics] notifyEnd]");
-    }
+    [self movePosition:properties];
```

Comment:
Since this is considered an end (ie. stop) event do we need the position here?

Reply:
Yes, you are right. MovePosition should not be called here.